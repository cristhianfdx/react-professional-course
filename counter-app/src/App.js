import React, { Fragment } from 'react';
import PropTypes from 'prop-types';

const App = ({ greeting, subtitle }) => {
  return (
    <Fragment>
      <h1>{greeting}</h1>
      <p>{subtitle}</p>
    </Fragment>
  );
};

App.propTypes = {
  greeting: PropTypes.string.isRequired,
};

App.defaultProps = {
  subtitle: 'Test',
};

export default App;
