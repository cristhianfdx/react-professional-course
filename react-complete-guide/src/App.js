import React, { Component } from 'react';
import Person from './Person/Person';
import './App.css';

class App extends Component {
  constructor() {
    super();
    this.state = {
      persons: [
        { name: 'Max', age: 28 },
        { name: 'Cristhian', age: 31 },
        { name: 'Stephanie', age: 26 },
      ],
      showPersons: false,
    };
  }

  switchNameHandler = (newName) => {
    this.setState({
      persons: [
        { name: newName, age: 28 },
        { name: 'Cristhian', age: 28 },
        { name: 'Stephanie', age: 27 },
      ],
    });
  };

  nameChangeHandler(event) {
    this.setState({
      persons: [
        { name: 'Max', age: 28 },
        { name: event.target.value, age: 31 },
        { name: 'Stephanie', age: 26 },
      ],
    });
  }

  togglePersonsHandler = () => {
    const doesShow = this.state.showPersons;
    this.setState({ showPersons: !doesShow });
  }

  render() {
    const style = {
      backgroundColor: 'white',
      font: 'inherit',
      border: '1px solid blue',
      padding: '8px',
      cursor: 'pointer',
    };

    return (
      <div className="App">
        <h1>Hi, I'm React App</h1>
        <button style={style} onClick={this.togglePersonsHandler}>
          Switch Name
        </button>
        {this.state.showPersons === true ? (
          <div>
            <Person name={this.state.persons[0].name} age={this.state.persons[0].age} />
            <Person
              name={this.state.persons[1].name}
              age={this.state.persons[1].age}
              click={this.switchNameHandler.bind(this, 'Cristhian!')}
              changed={this.nameChangeHandler.bind(this)}
            >
              My Hobbies: Racing
            </Person>
            <Person name={this.state.persons[2].name} age={this.state.persons[2].age} />
          </div>
        ) : null}
      </div>
    );
  }
}

// Functional Component

// const App = (props) => {
//   const [personsState, setPersonsState] = useState({
//     persons: [
//       { name: 'Max', age: 28 },
//       { name: 'Cristhian', age: 31 },
//       { name: 'Stephanie', age: 26 },
//     ],
//   });

//   const switchNameHandler = () => {
//     setPersonsState({
//       persons: [
//         { name: 'Lucy', age: 28 },
//         { name: 'Cristhian', age: 31 },
//         { name: 'Stephanie', age: 26 },
//       ],
//     });
//   };

//   return (
//     <div className="App">
//       <h1>Hi, I'm React App</h1>
//       <button onClick={switchNameHandler}>Switch Name</button>
//       <Person name={personsState.persons[0].name} age={personsState.persons[0].age} />
//       <Person name={personsState.persons[1].name} age={personsState.persons[1].age}>
//         My Hobbies: Racing
//       </Person>
//       <Person name={personsState.persons[2].name} age={personsState.persons[2].age} />
//     </div>
//   );
// };

export default App;
