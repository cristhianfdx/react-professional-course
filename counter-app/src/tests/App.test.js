import '@testing-library/jest-dom';

import { render } from '@testing-library/react';
import { shallow } from 'enzyme';
import React from 'react';

import App from '../App';

describe('App component tests', () => {
  test('should to show App component', () => {
    const greeting = 'Hi, I am Goku';
    const wrapper = shallow(<App greeting={greeting} />);

    expect(wrapper).toMatchSnapshot();
  });

  test('should to show subtitle', () => {
    const greeting = 'Hi, I am Goku';
    const subtitle = 'Ok Google!';
    const wrapper = shallow(<App greeting={greeting} subtitle={subtitle} />);

    const paragraph = wrapper.find('p').text();

    expect(paragraph).toBe(subtitle);
  });
});
