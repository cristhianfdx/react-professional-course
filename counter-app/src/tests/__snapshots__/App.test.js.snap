// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component tests should to show App component 1`] = `
<Fragment>
  <h1>
    Hi, I am Goku
  </h1>
  <p>
    Test
  </p>
</Fragment>
`;
